pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -96,7 +97,8 @@ func (r *AtlasDeploymentReconciler) advancedDeploymentIdle(
 		}
 	}
 
-	result := handleSearchIndexes(ctx, r.Client, deployment, project.ID())
+	searchService := searchindex.NewProductionAtlasSearch(ctx.SdkClient.AtlasSearchApi)
```

Comment:
Why doesn't this service depend on the client provider as the other?

Reply:
uhm, let me fix that

Reply:
Wait I get it now. This is using the provider indirectly, because it uses `ctx.SdkClient` which comes from the provider here:
https://github.com/mongodb/mongodb-atlas-kubernetes/blob/main/pkg/controller/atlasdeployment/atlasdeployment_controller.go#L153

```go
	atlasSdkClient, _, err := r.AtlasProvider.SdkClient(workflowCtx.Context, project.ConnectionSecretObjectKey(), log)
	if err != nil {
		result := workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
		workflowCtx.SetConditionFromResult(api.DeploymentReadyType, result)
		return result.ReconcileResult(), nil
	}
	workflowCtx.SdkClient = atlasSdkClient

```

BTW I think we should revisit this. There is **no** reason to compute this dependency per reconcile call.

Instead the reconciler could be set up with whatever interface implementations needed... which basically will mean killing the provider or at least using it at reconciliation build time, not at reconcile runtime.

Reply:
BTW I do agree that the naming `NewProductionAtlasSearch` is probably not the best and has started this confusion. But again, I do not think we should be calling this dependency injections here on each call.

Reply:
That would be great, but how are you going to get the credentials, usually defined per project, to set up the client?